diff --git a/gstreamer.py b/gstreamer.py
index 8e35db7..e8ed1a9 100644
--- a/gstreamer.py
+++ b/gstreamer.py
@@ -124,6 +124,13 @@ class Screencast(GObject.GObject):
             endx = startx + width - 1
             endy = starty + height - 1
 
+        scale = self.video_source['scale']
+        startx = startx * scale
+        starty = starty * scale
+        endx = endx * scale
+        endy = endy * scale
+
+
         #
         # H264 requirement is that video dimensions are divisible by 2.
         # If they are not, we have to get rid of that extra pixel.
diff --git a/prefs.py b/prefs.py
index 3859ad8..625592f 100644
--- a/prefs.py
+++ b/prefs.py
@@ -314,15 +314,19 @@ class hw:
 
             for i in range(self.default_screen.get_n_monitors()):
                 rect = self.default_screen.get_monitor_geometry(i)
-                self.logger.debug("  Monitor {0} - X: {1}, Y: {2}, W: {3}, H: {4}".format(i,
+                scale = self.default_screen.get_monitor_scale_factor(i)
+
+                self.logger.debug("  Monitor {0} - X: {1}, Y: {2}, W: {3}, H: {4}, scale: {5}".format(i,
                                                                                           rect.x,
                                                                                           rect.y,
                                                                                           rect.width,
-                                                                                          rect.height))
+                                                                                          rect.height,
+                                                                                          scale))
                 self.screens.append({"x": rect.x,
                                      "y": rect.y,
                                      "width": rect.width,
-                                     "height": rect.height})
+                                     "height": rect.height,
+                                     "scale": scale})
 
             if self.default_screen.get_n_monitors() > 1:
                 self.combined_screen = {"x": 0, "y": 0,
